dask/array/routines.py

```diff
@@ -404,6 +405,71 @@ def ediff1d(ary, to_end=None, to_begin=None):
     return r
 
 
+def _gradient_kernel(f, grad_varargs, grad_kwargs):
+    return np.gradient(f, *grad_varargs, **grad_kwargs)
```

Comment:
It wouldn't be possible. `map_overlap` only accepts one non-keyword argument, `depth`, and a couple option arguments.

It doesn't take any variadic argument and provides no real mechanism to forward variadic arguments, which leaves us doing this.

While `kwargs` is taken by `map_overlap`, it gets fed into `map_blocks` and can overlap with user arguments. So isn't that safe to rely on generally.

Reply:
> Would it be possible to change map_overlap to accept `*args, **kwargs`?

Not sure if this is what you were meaning, but opened issue ( https://github.com/dask/dask/issues/3437 ) to discuss some improvements to `map_blocks`/`map_overlap` that would improve usability.

Reply:
Yeah, I'd personally be in favor of changing map_overlap to accept `*args` and `**kwargs`.  The map_blocks and map_partitions (dataframes) do this and it seems to be well loved by users.  It might be valuable to be consistent across API functions like this.

Reply:
So the larger questions of how `map_overlap` behaves I think we can save for the other issue (already some good discussion in there). Is there anything else you need on this particular point?

Reply:
Nope.  It sounds like cleaning up this function is future work regardless of the way in which the map_overlap concludes.